### PR TITLE
add 3.1.2 TiDB release notes

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add TiDB 3.1.2 release notes.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3487
- Other reference link(s):

### CODE REVIEW

**Comment by scsldb** [comment]
LGTM

File: releases/release-3.1.2.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+category: Releases
+---
+
+# TiDB 3.1.2 Release Notes
+
+Release date: June 4, 2020
+
+TiDB version: 3.1.2
+
+TiUP version: 3.1.2
```
Comment (TomShawn) [inline releases/release-3.1.2.md:12]:
```suggestion
TiDB Ansible version: 3.1.2
```

**Comment by TomShawn** [comment]
@lonng PTAL the change of TiUP version in this PR, thanks!

**Pull request merged**